libkbfs/protected_context.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"sync"
+
+	"github.com/keybase/client/go/logger"
+	"golang.org/x/net/context"
+)
+
+type defaultContextReplacer struct{}
+
+func (defaultContextReplacer) maybeReplaceContext(context.Context) {}
+
+// protectedContexte is a construct that helps avoid unwanted change of context
+// when the context needs to be stored.
+type protectedContext struct {
+	mu  sync.Mutex
```

Comment:
Maybe `RWMutex`, since `context()` will be called way more than the other methods and one caller doesn't need to block the other.